### ISSUE TITLE
Unable to install freeform in pipeline

### ISSUE THREAD

**Issue by mariuswestgaard** [solspace/craft3-freeform#568]
Hi
I'm having issues with installing freeform from our pipeline. Actually, everything installs successfully during the pipeline and the composer install at the end, with no errors or warnings, but Craft displays this error when visiting the Freeform plugin dashboard in the Craft dashboard:

`500 Internal server error`

In our logs it says:
```
[error][yii\db\Exception] PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'forms.type' in 'field list' in /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/db/Command.php:1302
Stack trace:
#0 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/db/Command.php(1302): PDOStatement->execute()
#1 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/db/Command.php(1168): yii\db\Command->internalExecute('SELECT `forms`....')
#2 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/db/Command.php(410): yii\db\Command->queryInternal('fetchAll', NULL)
#3 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/db/Query.php(249): yii\db\Command->queryAll()
#4 /srv/app/vitec-production/htdocs/vendor/craftcms/cms/src/db/Query.php(152): yii\db\Query->all(NULL)
#5 /srv/app/vitec-production/htdocs/vendor/solspace/craft-freeform/packages/plugin/src/Services/FormsService.php(68): craft\db\Query->all()
#6 /srv/app/vitec-production/htdocs/vendor/solspace/craft-freeform/packages/plugin/src/Controllers/DashboardController.php(17): Solspace\Freeform\Services\FormsService->getAllForms()
#7 [internal function]: Solspace\Freeform\Controllers\DashboardController->actionIndex()
#8 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#9 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#10 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('index', Array)
#11 /srv/app/vitec-production/htdocs/vendor/craftcms/cms/src/web/Application.php(293): yii\base\Module->runAction('freeform/dashbo...', Array)
#12 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('freeform/dashbo...', Array)
#13 /srv/app/vitec-production/htdocs/vendor/craftcms/cms/src/web/Application.php(278): yii\web\Application->handleRequest(Object(craft\web\Request))
#14 /srv/app/vitec-production/htdocs/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#15 /srv/app/vitec-production/htdocs/web/index.php(21): yii\base\Application->run()
#16 {main}
```

**Steps to reproduce**

1. Install freeform locally
2. Deploy to server (rsync), and run `composer install` as final step in deploy pipeline

**Expected behavior**
Expected composer to install the plugin, and migrate our database accordingly, but the database seems to not be fully migrated, as we see all the freeform-tables, but not the "type" column.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.51
- Freeform Version: 3.13.22.1
- Freeform Edition: lite
- Fresh Install or Upgrade: Fresh install

**Additional context**
Have tried running `composer update` in ssh. Have also removed vendor-folder and reinstalled with composer. Also tried uninstalling and requiring in ssh using composer. Have also tested removing all caches, but all with same result.

**Comment by kjmartens** [comment]
Hi @mariuswestgaard,

Can you have a look at this similar issue here:
https://github.com/solspace/craft-freeform/issues/468#issuecomment-1275686764

> Yes that's a risky thing to do. Might I recommend `composer install`ing directly on the server instead? `composer install` is meant exactly for that scenario, it installs all of the packages at the versions in the `composer.lock` file, whereas `composer update` updates all dependencies to the latest versions generating a new lock file.
>
> `rsync` could potentially skip overwriting files based on timestamps which might cause issues, as well as not removing files as you mentioned.

It might be that your existing flow could run you into an issue like this occasionally... if absolutely necessary, you can probably continue to use your current flow, but if an issue arises like this, `composer install`ing directly on the server instead should resolve the issue. 🙂 

Additionally, `composer update` doesn't update the Freeform tables, as Craft handles that. By looking at the migrations files, is it possible that you pushed your DB changes as well, which would then have the migration entries that craft generates when it updates?

To clarify, if you install the newest version locally, it would create entries for all migrations and not execute them at all, so if you then push those DB values to the server, it would think that the migrations have already been executed (but they wouldn't be).

**Comment by mariuswestgaard** [comment]
Hi @kjmartens 
Thanks for helping out.

I tried `composer install`ing directly on the server, but it still wouldn't fix the problem. I just found out that our database had the freeform tables existing before we installing the latest version, meaning we've already had the plugin installed earlier on a lower version (but it was uninstalled at some point). So, what worked was to install the version that previously was installed, and then go on and update the plugin, rather than installing the latest version.
